service/api/client/client.go

```diff
@@ -0,0 +1,282 @@
+}
+
+func (c *Client) CreateThread(ctx context.Context, id thread.ID) (info thread.Info, err error) {
+	resp, err := c.c.CreateThread(ctx, &pb.CreateThreadRequest{
+		ThreadID: id.Bytes(),
+	})
+	if err != nil {
+		return
+	}
+	return threadInfoFromPbThread(resp)
+}
+
+func (c *Client) AddThread(ctx context.Context, addr ma.Multiaddr, opts ...core.AddOption) (info thread.Info, err error) {
+	args := &core.AddOptions{}
+	for _, opt := range opts {
+		opt(args)
+	}
+	resp, err := c.c.AddThread(ctx, &pb.AddThreadRequest{
+		Addr:      addr.Bytes(),
+		ReadKey:   args.ReadKey.Bytes(),
+		FollowKey: args.FollowKey.Bytes(),
+	})
+	if err != nil {
+		return
+	}
+	return threadInfoFromPbThread(resp)
+}
+
+func (c *Client) GetThread(ctx context.Context, id thread.ID) (info thread.Info, err error) {
+	resp, err := c.c.GetThread(ctx, &pb.GetThreadRequest{
+		ThreadID: id.Bytes(),
+	})
+	if err != nil {
+		return
+	}
+	return threadInfoFromPbThread(resp)
+}
+
+func (c *Client) PullThread(ctx context.Context, id thread.ID) error {
+	_, err := c.c.PullThread(ctx, &pb.PullThreadRequest{
+		ThreadID: id.Bytes(),
+	})
+	return err
+}
+
+func (c *Client) DeleteThread(ctx context.Context, id thread.ID) error {
+	_, err := c.c.DeleteThread(ctx, &pb.DeleteThreadRequest{
+		ThreadID: id.Bytes(),
+	})
+	return err
+}
+
+func (c *Client) AddFollower(ctx context.Context, id thread.ID, paddr ma.Multiaddr) (peer.ID, error) {
+	resp, err := c.c.AddFollower(ctx, &pb.AddFollowerRequest{
+		ThreadID: id.Bytes(),
+		Addr:     paddr.Bytes(),
+	})
+	if err != nil {
+		return "", err
+	}
+	return peer.IDFromBytes(resp.PeerID)
+}
+
+func (c *Client) AddRecord(ctx context.Context, id thread.ID, body format.Node) (core.ThreadRecord, error) {
+	resp, err := c.c.AddRecord(ctx, &pb.AddRecordRequest{
+		ThreadID: id.Bytes(),
+		Body:     body.RawData(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	return threadRecordFromPbNewRecord(resp)
+}
+
+func (c *Client) GetRecord(ctx context.Context, id thread.ID, rid cid.Cid) (core.Record, error) {
+	resp, err := c.c.GetRecord(ctx, &pb.GetRecordRequest{
+		ThreadID: id.Bytes(),
+		RecordID: rid.Bytes(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	return recordFromPbRecord(resp.Record)
+}
+
+func (c *Client) Subscribe(ctx context.Context, opts ...core.SubOption) (<-chan core.ThreadRecord, error) {
+	args := &core.SubOptions{}
+	for _, opt := range opts {
+		opt(args)
+	}
+	threadIDs := make([][]byte, len(args.ThreadIDs))
+	for i, id := range args.ThreadIDs {
+		threadIDs[i] = id.Bytes()
+	}
+	stream, err := c.c.Subscribe(ctx, &pb.SubscribeRequest{
+		ThreadIDs: threadIDs,
+	})
+	if err != nil {
+		return nil, err
+	}
+	channel := make(chan core.ThreadRecord)
+	go func() {
+		defer close(channel)
+		for {
+			resp, err := stream.Recv()
+			if err == io.EOF {
+				return
+			}
+			if err != nil {
+				stat := status.Convert(err)
+				if stat.Code() != codes.Canceled {
+					log.Fatalf("error in subscription stream: %v", err)
```

Comment:
Maybe not fatal?
I think there're some more there, so may apply to other lines.

Reply:
Sure, sounds good